File: 1207.py
from collections import defaultdict

class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        count = defaultdict(lambda : 0)
        for num in arr:
            count[num] += 1
        counts = set(count.values())
        if(len(count) == len(counts)):
            return True
        return False

File: 1207.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> count;
        unordered_set<int> counts;

        for(int i = 0; i < arr.size(); i++) {
            count[arr[i]]++;
        }
        for(auto it = count.begin(); it != count.end(); it++) {
            if(counts.find(it->second) == counts.end()) {
                counts.insert(it->second);
            } else {
                return false;
            }
        }
        return true;
    }
};